### PR TITLE
Add DSA Array Questions (Easy, Medium, Hard)

### LINKED ISSUE #1244: [Feature]: Add DSA Array Questions (Easy, Medium, Hard)
### Feature Name

Add DSA Array Questions (Easy, Medium, Hard)

### Feature Description

This feature will add 6 array-based DSA questions to the arrays folder which will be located in  algo/java folder. The questions will be divided into three difficulty levels: 2 easy, 2 medium, and 2 hard. These questions will help users practice and improve their problem-solving skills in array-related problems, providing a structured approach to learning.

### Motivation

Adding a diverse set of array questions will enhance the utility of the Algo repository, enabling users to practice arrays at different difficulty levels. It serves as a valuable resource for those looking to strengthen their understanding of Data Structures and Algorithms (DSA), particularly arrays.

### Implementation Suggestions (Optional)

Create a new folder: algo/java/arrays/
Add 2 easy, 2 medium, and 2 hard questions .
Consider adding a README.md in the arrays folder to give users a brief explanation and a list of questions.

### Feature Type

New Algorithm

### Does this feature require additional resources?

- [X] Additional Documentation
- [ ] Third-Party Libraries
- [ ] UI/UX Design
- [ ] Other

### References (Optional)

Add links to relevant DSA resources such as LeetCode or GeeksforGeeks for array-based questions.

### PR DESCRIPTION
**Pull request by Shreya7tripathy**
### Description
Added an `arrays` folder in the `java` folder containing easy, medium, and hard questions. This change contributes to improving the problem set available for Java enthusiasts and includes solutions categorized by difficulty levels.

Fixes #1244  issue.

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [x] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
please carefully read our https://github.com/ajay-dhangar/algo/blob/main/CONTRIBUTING.md

**Comment by Shreya7tripathy** [comment]
> please carefully read our https://github.com/ajay-dhangar/algo/blob/main/CONTRIBUTING.md

Ok I will make the necessary changes soon

**Comment by Shreya7tripathy** [comment]
Done changes. Please check @ajay-dhangar

File: java/Arrays/1.Easy/Find missing number.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
+---
+id: <find-missing-number>
+title: <Find the Missing Number in Array>
+sidebar_label: <Find Missing Number in Array>
+sidebar_position: <1>
+description: <Find the missing number in an array of size N containing numbers from 1 to N.>
+tags: [<Array>, <XOR>, <DSA>]
+---
```
Comment (ajay-dhangar) [inline java/Arrays/1.Easy/Find missing number.md]: replace 
```md
---
id: <find-missing-number>
title: <Find the Missing Number in Array>
sidebar_label: <Find Missing Number in Array>
sidebar_position: <1>
description: <Find the missing number in an array of size N containing numbers from 1 to N.>
tags: [<Array>, <XOR>, <DSA>]
---
```
with
```md
---
id: find-missing-number
title: Find the Missing Number in Array
sidebar_label: Find Missing Number in Array
sidebar_position: 1
description: Find the missing number in an array of size N containing numbers from 1 to N.
tags: [array, xor, dsa]
---
```

File: java/Arrays/1.Easy/Remove Duplicates from Sorted Array.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
+---
+id: <remove-duplicates-sorted-array>
+title: <Remove Duplicates in-place from Sorted Array>
+sidebar_label: <Remove Duplicates in-place>
+sidebar_position: <1>
+description: <Given a sorted array, remove the duplicates in-place and return the new length.>
+tags: [<Array>, <In-place>, <DSA>]
+---
+
```
Comment (ajay-dhangar) [inline java/Arrays/1.Easy/Remove Duplicates from Sorted Array.md]: same mistake so, can you resolve it

File: java/Arrays/2.Medium/Longest Subarray with sum K.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
+---
+id: <longest-subarray-sum-k>
+title: <Find Longest Subarray with Sum k>
+sidebar_label: <Longest Subarray with Sum k>
+sidebar_position: <1>
+description: <Find the length of the longest subarray whose sum equals k.>
+tags: [<Array>, <Hashing>, <DSA>]
+---
```
Comment (ajay-dhangar) [inline java/Arrays/2.Medium/Longest Subarray with sum K.md]: ❌

File: java/Arrays/3.Hard/Maximum Product Subarray.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
+---
+id: <maximum-product-subarray>
+title: <Maximum Product Subarray in an Array>
+sidebar_label: <Maximum Product Subarray>
+sidebar_position: <1>
+description: <Find the maximum product of a contiguous subarray within an array containing both negative and positive integers.>
+tags: [<Dynamic Programming>, <Array>]
+---
```
Comment (ajay-dhangar) [inline java/Arrays/3.Hard/Maximum Product Subarray.md]: ❌

File: java/Arrays/3.Hard/Reverse pairs.md [hunk 5]
```diff
@@ -0,0 +1,139 @@
+---
+id: <reverse-pairs>
+title: <Count Reverse Pairs>
+sidebar_label: <Count Reverse Pairs>
+sidebar_position: <1>
+description: <Count the number of reverse pairs in an array where i < j and arr[i] > 2 * arr[j].>
+tags: [<Merge Sort>, <Array>]
+---
```
Comment (ajay-dhangar) [inline java/Arrays/3.Hard/Reverse pairs.md]: ❌

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
Update your code based on suggestions

**Comment by Shreya7tripathy** [comment]
Thanks for guiding me. Sure I will do it now . @ajay-dhangar

File: java/Arrays/3.Hard/Reverse pairs.md [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+id: reverse-pairs
+title: Count Reverse Pairs
+sidebar_label: Count Reverse Pairs
+sidebar_position: 1
+description: Count the number of reverse pairs in an array where i < j and arr[i] > 2 * arr[j].
```
Comment (ajay-dhangar) [inline java/Arrays/3.Hard/Reverse pairs.md]: ⁉️

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
Recheck your code

File: java/Arrays/3.Hard/Reverse pairs.md (hunk 6 above)
Comment (Shreya7tripathy) [reply java/Arrays/3.Hard/Reverse pairs.md]: > ⁉️

can you please elaborate your concern?

Comment (ajay-dhangar) [reply java/Arrays/3.Hard/Reverse pairs.md]: In the description, you should use only text content (as a string). Do not use language keywords and syntax.

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**